### PR TITLE
Simplify colormap plotting

### PR DESCRIPTION
**Pull request by QuLogic**
All the special cases other than Greys are just the size of the colormap themselves. Also, colormaps are at most 256 elements long, so using `n=512` is probably not necessary.

Also, use the same algorithm as the [Specifying colors tutorial](https://matplotlib.org/stable/users/explain/colors/colors.html#comparison-between-x11-css4-and-xkcd-colors) to choose text colour over colour cycles.

### CODE REVIEW

**Comment by rougier** [comment]
Didn't know about the `cmap.N`... Looks good to me.

File: scripts/colormaps.py [hunk 1]
```diff
@@ -35,36 +35,14 @@
...
-
-    if cmap in ['tab10']: n = 10
-    if cmap in ['tab20', 'tab20b', 'tab20c']: n = 20
-    if cmap in ['Pastel1', 'Set1']: n = 9
-    if cmap in ['Pastel2', 'Accent', 'Dark2', 'Set2']: n = 8
-    if cmap in ['Set3']: n = 12
-    if cmap in ['Greys']: n = 11
+for name in cmaps:
+    cmap = plt.get_cmap(name)
+    n = cmap.N if name != 'Greys' else 11
```
Comment (timhoffm) [inline scripts/colormaps.py:40]: IMHO "Greys" should not be resampled. They are a quasi-continuous colormap. https://matplotlib.org/devdocs/gallery/color/colormap_reference.html#sphx-glr-gallery-color-colormap-reference-py

This also simplifies the code further in that we don't need to `get_cmap()` the instances but can simply pass `cmap=Name`.

Comment (timhoffm) [reply scripts/colormaps.py:40]: I don’t see a reason why one would do that. In contrast:

- Resampled is not how the color bar looks / how color mapping is performed for `camp=“Greys”`.
- This is a special and different treatment than for all other color maps.

Comment (rougier) [reply scripts/colormaps.py:40]: Not sure but I think the idea was to match the grey color definition but now I see it brings confusion by suggesting grey colormap is discrete. So I think you're right (no resample)

**Review by timhoffm — APPROVED** [review]

**Comment by timhoffm** [comment]
Do we have a one or two reviewer policy on th cheatsheets?

**Pull request merged**